File: 2526-find-consecutive-integers-from-a-data-stream/2526-find-consecutive-integers-from-a-data-stream.cs
public class DataStream {

    private int _value, _k, canAdd = 0;
    
    public DataStream(int value, int k) {
         _value = value;
         _k = k;
     
    }
    
    public bool Consec(int num) {
        if (num == _value){
            canAdd++;
        }else{
            canAdd = 0;
        }
        
        
        if (canAdd >= _k){
            return true;
        }
        return false;
    }
}

/**
 * Your DataStream object will be instantiated and called as such:
 * DataStream obj = new DataStream(value, k);
 * bool param_1 = obj.Consec(num);
 */

File: 2526-find-consecutive-integers-from-a-data-stream/2526-find-consecutive-integers-from-a-data-stream.py
class DataStream(object):

    def __init__(self, value, k):
        """
        :type value: int
        :type k: int
        """
        
        self.value = value
        self.k = k
        
        self.canAdd = 0
        
        

    def consec(self, num):
        """
        :type num: int
        :rtype: bool
        """
        
        if num == self.value:
            self.canAdd += 1
        else:
            self.canAdd = 0
            
        if self.canAdd >= self.k:
            return True
        else:
            return False
        
        


# Your DataStream object will be instantiated and called as such:
# obj = DataStream(value, k)
# param_1 = obj.consec(num)